HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiVariable.java

```diff
@@ -0,0 +1,180 @@
+  int variableLength = 0;
+
+  System.arraycopy(varibaleData, 0, guid, 0, UefiConstants.SIZE_16);
+  uefiGuid = new UefiGuid(guid);
+  System.arraycopy(varibaleData, UefiConstants.SIZE_16, nameLength, 0, UefiConstants.SIZE_8);
+  int nlength = HexUtils.leReverseInt(nameLength);
+  System.arraycopy(varibaleData, UefiConstants.OFFSET_24, dataLength, 0, UefiConstants.SIZE_8);
+  nameTemp = new byte[nlength * UefiConstants.SIZE_2];
+
+  System.arraycopy(varibaleData, UefiConstants.OFFSET_32,
+                   nameTemp, 0, nlength * UefiConstants.SIZE_2);
+  byte[] name1 = UefiDevicePath.convertChar16tobyteArray(nameTemp);
+  name = new byte[nlength];
+  System.arraycopy(name1, 0, name, 0, nlength);
+  variableLength = HexUtils.leReverseInt(dataLength);
+  uefiVaribelData = new byte[variableLength];
+  System.arraycopy(varibaleData, UefiConstants.OFFSET_32
+                   + nlength * UefiConstants.SIZE_2, uefiVaribelData, 0, variableLength);
+  varName = new String(name, "UTF-8");
+  String tmpName = varName;
+  if (varName.contains("Boot00")) {
+      tmpName = "Boot00";
+  }
+  switch (tmpName) {
+      case "PK":          processSigList(uefiVaribelData); break;
+      case "KEK":         processSigList(uefiVaribelData); break;
+      case "db":          processSigList(uefiVaribelData); break;
+      case "dbx":         processSigList(uefiVaribelData); break;
+      case "Boot00":      bootv = new UefiBootVariable(uefiVaribelData); break;
+      case "BootOrder":   booto = new UefiBootOrder(uefiVaribelData); break;
+      case "SecureBoot":  sb = new UefiSecureBoot(uefiVaribelData); break;
+      default:
+      }
+}
+
+/**
+ * The GUID is a globally unique identifier assigned to this UEFI variable.
+ * UEFI variable specific GUIDs are specified in the UEFI specification.
+ * @return UEFI Variable GUID.
+ */
+public UefiGuid getEfiVarGuid() {
+    return uefiGuid;
+}
+
+/**
+ * Returns a human readable Name for this UEFI Variable (e.g. SecureBoot).
+ * @return the UEFI Variable name assigned to this variable.
+ */
+public String getEfiVarName() {
+   return varName;
+}
+
+/**
+ * Returns a arrayList of UefiSignatureList (Certificates or hashes) held in the Event Log.
+ * @return and array list of UEFI defined Signature Lists.
+ */
+public ArrayList<UefiSignatureList> getEFISignatureList() {
+  return certSuperList;
+}
+
+/**
+ * Processes the data as a UEFI defined Signature List.
+ * @param data the bye array holding the Signature List.
+ * @throws CertificateException If there a problem parsing the X509 certificate.
+ * @throws NoSuchAlgorithmException if there's a problem hashing the certificate.
+ * @throws IOException If there's a problem parsing the signature data.
+ */
+private void processSigList(final byte[] data)
+        throws CertificateException, NoSuchAlgorithmException, IOException {
+  ByteArrayInputStream certData = new ByteArrayInputStream(data);
+  while (certData.available() > 0) {
+     UefiSignatureList list;
```

Comment:
You can move this statement outside of the loop

Reply:
ditto